packages/repo-tools/src/generate-configs.mts

```diff
@@ -141,38 +163,101 @@ async function main(): Promise<void> {
     const hyphens = '-'.repeat(35 - Math.ceil(name.length / 2));
     console.log(chalk.blueBright(`\n${hyphens} ${name}.ts ${hyphens}`));
 
+    const config = getConfig();
+
+    //
+    // 1. Classic Config - written to the eslint-plugin package
+    // These configs are just JSON blobs that we write as TS files
+    //
+
     // note: we use `export =` because ESLint will import these configs via a commonjs import
-    const code = [
+    const classicCode = [
       "import type { ClassicConfig } from '@typescript-eslint/utils/ts-eslint';",
       '',
-      `export = ${JSON.stringify(getConfig())} satisfies ClassicConfig.Config;`,
+      `export = ${JSON.stringify(config)} satisfies ClassicConfig.Config;`,
     ].join('\n');
-    const configStr = await prettier.format(addAutoGeneratedComment(code), {
+    const classicConfigStr = await prettier.format(
+      addAutoGeneratedComment(classicCode),
+      {
+        parser: 'typescript',
+        ...prettierConfig,
+      },
+    );
+    fs.writeFileSync(
+      path.join(PACKAGES_ESLINT_PLUGIN, 'src', 'configs', `${name}.ts`),
+      classicConfigStr,
+    );
+
+    //
+    // 2. Flat Config - written to the core package
+    // These configs are actual TS modules that import other configs
+    //
+    const flatCode: string[] = [
+      ...AUTO_GENERATED_COMMENT_LINES,
+      "import type { FlatConfig } from '@typescript-eslint/utils/ts-eslint';",
+      '',
+    ];
+    const flatExtends: string[] = [];
+    const flatConfig: FlatConfig.Config = {
+      rules: config.rules,
+    };
+    if (config.extends) {
+      for (const extendPath of config.extends) {
+        const config = EXTENDS_MODULES.find(
+          mod => mod.packageRelativePath === extendPath,
+        );
+        if (config == null) {
+          throw new Error("Couldn't find config");
+        }
+        flatCode.push(
+          `import ${config.name} from '${config.moduleRelativePath}';`,
+        );
+        flatExtends.push(config.name);
+      }
+      flatCode.push('');
+    }
+    if (config.parserOptions) {
+      flatConfig.languageOptions ??= {};
+      flatConfig.languageOptions.parserOptions = config.parserOptions;
+    }
+
+    const flatConfigJson = JSON.stringify(flatConfig);
+    if (flatExtends.length > 0) {
+      flatCode.push(
+        'export default (plugin: FlatConfig.Plugin, parser: FlatConfig.Parser): FlatConfig.ConfigArray => [',
+        ...flatExtends.map(ext => `${ext}(plugin, parser),`),
+        flatConfigJson,
+        '];',
+      );
+    } else {
+      flatCode.push(
+        `export default (_plugin: FlatConfig.Plugin, _parser: FlatConfig.Parser): FlatConfig.Config => (${flatConfigJson});`,
```

Comment:
[Style] If they're unused, why bother including them?

```suggestion
        `export default (): FlatConfig.Config => (${flatConfigJson});`,
```

Here, and the unused import above?

Reply:
cos it sets things up to have a consistent API shape - which is nice in case we do change things in future.

I.e. it means we can blindly write
```
configs: {
    all: allConfig(plugin, parser),
    base: baseConfig(plugin, parser),
    disableTypeChecked: disableTypeCheckedConfig(plugin, parser),
    eslintRecommended: eslintRecommendedConfig(plugin, parser),
    recommended: recommendedConfig(plugin, parser),
    recommendedTypeChecked: recommendedTypeCheckedConfig(plugin, parser),
    strict: strictConfig(plugin, parser),
    strictTypeChecked: strictTypeCheckedConfig(plugin, parser),
    stylistic: stylisticConfig(plugin, parser),
    stylisticTypeChecked: stylisticTypeCheckedConfig(plugin, parser),
}
```

rather than having to maintain which ones get args and which don't.